packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -502,7 +515,37 @@ const InputBase = React.forwardRef(function InputBase(inProps, ref) {
         {...other}
         className={clsx(classes.root, rootProps.className, className)}
       >
-        {startAdornment}
+        {(() => {
+          if (React.isValidElement(startAdornment)) {
+            return React.cloneElement(startAdornment, {
+              ref: (newRef) => {
+                startAdornmentRefs.current[0] = newRef;
+              },
+            });
+          }
+          if (Array.isArray(startAdornment)) {
+            return startAdornment.map((element, i) =>
+              React.cloneElement(element, {
+                ref: (newRef) => {
+                  startAdornmentRefs.current[i] = newRef;
+                },
+              }),
+            );
+          }
+          if (startAdornment) {
+            return (
+              <div
+                ref={(newRef) => {
+                  startAdornmentRefs.current[0] = newRef;
+                }}
+              >
+                {startAdornment}
+              </div>
+            );
+          }
```

Comment:
IMHO, this is a deal-breaker. Could we find a very simple solution to this problem? It's not really important, so doesn't deserve the complexity.

Reply:
@oliviertassinari All right, I can change it so that the width is provided by a prop. Should it be a prop of FormControl, or InputAdornment, or both?

Reply:
I think that a prop for FormControl would be enough. What do you think?

Reply:
@oliviertassinari Yeah, that seems fine. I'll go ahead with making the changes.